### PR TITLE
Changes from the MAD series on WorkManager

### PR DESCRIPTION
**Pull request by keyboardsurfer**
Contains several changes that were made to improve the codebase throughout the MAD series.

### CODE REVIEW

File: WorkManagerSample/app/src/main/java/com/example/background/FilterViewModel.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class FilterViewModel(application: Application) : ViewModel() {
     private val workManager = WorkManager.getInstance(application)
 
     internal val workInfo =
-        workManager.getWorkInfosByTagLiveData(Constants.TAG_OUTPUT).map { it[0] }
```
Comment (tikurahul) [inline WorkManagerSample/app/src/main/java/com/example/background/FilterViewModel.kt:35]: OOC, why this change? Was it unclear before ?

File: WorkManagerSample/lib/src/androidTest/java/com/example/background/workers/filters/BaseFilterWorkerTest.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+class BaseFilterWorkerTest {
+
+    private lateinit var worker: TestBaseFilterWorker
+    private val context: Context = ApplicationProvider.getApplicationContext()
+
+    @Test
+    fun testFiltering_withExistingUri() {
+        val worker = TestWorkerBuilder<TestBaseFilterWorker>(
+            context,
+            newSingleThreadExecutor(),
```
Comment (tikurahul) [inline WorkManagerSample/lib/src/androidTest/java/com/example/background/workers/filters/BaseFilterWorkerTest.kt:45]: `newSingleThreadExecutor` should be `shutdown`, on `tearDown()`

**Review by tikurahul — APPROVED** [review]

File: WorkManagerSample/app/src/main/java/com/example/background/FilterViewModel.kt (hunk 1 above)
Comment (keyboardsurfer) [reply WorkManagerSample/app/src/main/java/com/example/background/FilterViewModel.kt:35]: Encountered an issue when the list was empty. Changed it back to use `map` again.
